File: EngimonLiar.cpp
#include "Species.hpp"
#include "Element.hpp"
#include "Engimon.hpp"
#include "EngimonLiar.hpp"
#include <bits/stdc++.h>
using namespace std;

EngimonLiar::EngimonLiar(const Species& sp, Position pos, int level) : Engimon(sp.getName(), sp, level), position(pos){}

File: EngimonLiar.java
public class EngimonLiar extends Engimon implements Creature {
    private Position position;

    EngimonLiar(String name, Species species, int level, Position p) {
        super(name, species, level);
        this.position = p;
    }

    @Override
    public Position getPosition() { return position; }

    @Override
    public void setPosition(Position p) {
        this.position = p;
    }

    public void increaseLevel() {
        this.addExp(Engimon.EXP_PER_LEVEL);
    }
}
